### ISSUE TITLE
test failure following plugin2 changes in #14

### ISSUE THREAD

**Issue by SysPete** [ironcamel/Dancer2-Plugin-DBIC#15]
Switched to a different system and I'm getting failures in t/02-multiple-schemas.t with this :unamused: 

```
#   Failed test 'Default schema'
#   at t/02-multiple-schemas.t line 79.
# died: The schema default is not configured at /home/apm/perl5/perlbrew/perls/perl-5.20.3/lib/site_perl/5.20.3/DBICx/Sugar.pm line 52.
```
No idea how this worked on main work system but is failing on the laptop. Will get it fired up and see if I can find out what is happening.

**Comment by SysPete** [comment]
My main test env was borked which is why I didn't pick up on this earlier.

@xsawyerx seems plugin2 lazy-loads plugin config which becomes immutable so later plugin config changes do *not* change config (accessed via $plugin->config). Can you please confirm that this is correct and expected?

**Comment by SysPete** [comment]
@ironcamel so this is due to plugin2 having a lazily loaded immutable config attribute. This is not going to change before plugin2 hits CPAN in the next Dancer2 release so I'll create a PR which splits 02-multiple-schemas.t into 2 tests.

**Comment by SysPete** [comment]
This is closed by PR #16 so I'm closing this issue.
